### PR TITLE
SALTO-2799: Salesforce partial fetch dependencies

### PR DESCRIPTION
**Pull request by tamtamirr**
Added support for MetadataTypes with dependencies in Salesforce fetch targets (partial fetch).

---

_Additional context for reviewer_

The code is fully typed to avoid any errors. Note that some of the checks are handled in UTs due to limitations of the language (Like duplications between metadata types with dependencies and without)

---

_Release Notes_: 
_None_

---

_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      .value()
+  )
+  const isSupportedMetadataType = (typeName: string): boolean => (
+    (SUPPORTED_METADATA_TYPES as ReadonlyArray<string>).includes(typeName)
+  )
+  const isUnsupportedMetadataType = (typeName: string): boolean => (
+    !isSupportedMetadataType(typeName)
+  )
+
+  // afterAll(() => {
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts:54]: Let's remove this

File: packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  //   jest.resetAllMocks()
+  // })
+
+  it.each([
+    ['METADATA_TYPES_WITHOUT_DEPENDENCIES', METADATA_TYPES_WITHOUT_DEPENDENCIES as ReadonlyArray<string>],
+    ['METADATA_TYPES_WITH_DEPENDENCIES', METADATA_TYPES_WITH_DEPENDENCIES as ReadonlyArray<string>],
+    ['EXCLUDED_METADATA_TYPES', EXCLUDED_METADATA_TYPES as ReadonlyArray<string>],
+    ['CUSTOM_OBJECT_DEPENDENCIES', CUSTOM_OBJECT_DEPENDENCIES as ReadonlyArray<string>],
+    ['WORKFLOW_DEPENDENCIES', WORKFLOW_DEPENDENCIES as ReadonlyArray<string>],
+  ])('%p should not contain duplicates', (__, array) => {
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts:64]:
```suggestion
    ['METADATA_TYPES_WITHOUT_DEPENDENCIES', METADATA_TYPES_WITHOUT_DEPENDENCIES as ReadonlyArray<string>],
    ['METADATA_TYPES_WITH_DEPENDENCIES', METADATA_TYPES_WITH_DEPENDENCIES as ReadonlyArray<string>],
    ['EXCLUDED_METADATA_TYPES', EXCLUDED_METADATA_TYPES as ReadonlyArray<string>],
    ['CUSTOM_OBJECT_DEPENDENCIES', CUSTOM_OBJECT_DEPENDENCIES as ReadonlyArray<string>],
    ['WORKFLOW_DEPENDENCIES', WORKFLOW_DEPENDENCIES],
  ] as ReadonlyArray<string>][])('%p should not contain duplicates', (__, array) => {
```

**Review by amitgoldarb — APPROVED** [review]
LGTM! Discussed f2f
Thanks for this

File: packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/test/fetch_profile/metadata_types.test.ts]: v

**Pull request merged**